src/metabase/models/database.clj

```diff
@@ -314,6 +322,33 @@
             driver.u/default-sensitive-fields))
       driver.u/default-sensitive-fields))
 
+(defn- redact-db-details [db]
+  (if (not (mi/can-write? db))
+    (u/assoc-existing db :details protected-db-details)
+    (m/update-existing db :details (fn [details]
+                                     (reduce
+                                      #(m/update-existing %1 %2 (constantly protected-password))
+                                      details
+                                      (sensitive-fields-for-db db))))))
+
+(defn- redact-settings [settings]
+  (if-not (or (nil? settings) (map? settings))
```

Comment:
when `settings` is not a map?

Reply:
This was just preserving existing code. I suspect if it shouldn't be, and we should probably fail this when querying the data.